overlord/ifacestate/helpers.go

```diff
@@ -102,14 +108,26 @@ func (m *InterfaceManager) addSnaps() error {
 	return nil
 }
 
-// regenerateAllSecurityProfiles will regenerate the security profiles
-// for apparmor and seccomp. This is needed because:
-// - for seccomp we may have "terms" on disk that the current snap-confine
-//   does not understand (e.g. in a rollback scenario). a refresh ensures
-//   we have a profile that matches what snap-confine understand
-// - for apparmor the kernel 4.4.0-65.86 has an incompatible apparmor
-//   change that breaks existing profiles for installed snaps. With a
-//   refresh those get fixed.
+func (m *InterfaceManager) profilesNeedRegeneration() bool {
+	currentSystemKey := interfaces.SystemKey()
+	if currentSystemKey == "" {
+		logger.Noticef("no system key, forcing re-generation of security profiles")
+		return true
+	}
+
+	onDiskSystemKey, err := ioutil.ReadFile(dirs.SnapSystemKeyFile)
+	if os.IsNotExist(err) {
+		return true
+	}
```

Comment:
If you wanted, you could combine these since they are effectively the same thing:
```
currentSystemKey := interfaces.SystemKey()
onDiskSystemKey, err := ioutil.ReadFile(dirs.SnapSystemKeyFile)
if currentSystemKey == "" || os.IsNotExist(err) {
     logger.Noticef("no system key, forcing re-generation of security profiles")
     return true
}
```

Not blocking on that; the code is fine as is.